source/list_utils.c

```diff
@@ -35,13 +40,17 @@ int aws_cryptosdk_transfer_list(struct aws_array_list *dest, struct aws_array_li
 typedef int (*clone_item_fn)(struct aws_allocator *, void *, const void *);
 typedef void (*clean_up_item_fn)(void *);
 
-static int list_copy_all(
+/*static*/ int list_copy_all(
     struct aws_allocator *alloc,
     struct aws_array_list *dest,
     const struct aws_array_list *src,
     clone_item_fn cloner,
     clean_up_item_fn cleaner) {
-    assert(dest->item_size == src->item_size);
+    AWS_ERROR_PRECONDITION(aws_allocator_is_valid(alloc));
```

Comment:
(picky) you could move this validity check together with the others and leave `src != dest` first.

Reply:
I liked the idea of checking in the same order as the function params.